crates/polars-plan/src/logical_plan/optimizer/predicate_pushdown/mod.rs

```diff
@@ -260,7 +260,13 @@ impl<'a> PredicatePushDown<'a> {
                 file_options: options,
                 output_schema
             } => {
-                let local_predicates = partition_by_full_context(&mut acc_predicates, expr_arena);
+                let mut local_predicates = partition_by_full_context(&mut acc_predicates, expr_arena);
+                if let Some(ref row_count) = options.row_count{
+                    let mut row_count_predicates = transfer_to_local_by_name(expr_arena, &mut acc_predicates, |name| {
+                        name.as_ref() == &row_count.name
+                    });
+                    local_predicates.append(&mut row_count_predicates);
```

Comment:
I think `extend_from_slice` is better here. That has less overhead as it can just memcopy and doesn't have to pop from the right hand side.